src/nerve/server/handlers/node_lifecycle_handler.py

```
@@ -374,6 +378,38 @@ async def get_node(self, params: dict[str, Any]) -> dict[str, Any]:
+
+        Returns:
+            Dict with "tools" list containing tool definitions.
+        """
+        from nerve.core.nodes.tools import ToolCapable
+
+        session = self.session_registry.get_session(params.get("session_id"))
+        node_id = self.validation.require_param(params, "node_id")
+        node = self.validation.get_node(session, node_id)
+
+        if not isinstance(node, ToolCapable):
+            raise ValueError(f"Node '{node_id}' does not support tools")
```

Ruff (0.14.10)

Prefer `TypeError` exception for invalid type

(TRY004)